third_party/move/move-compiler-v2/tests/recursive-instantiation-checker/v1-tests/complex_1.exp

```diff
@@ -0,0 +1,37 @@
+
+Diagnostics:
+error: cyclic type instantiation
+   ┌─ tests/recursive-instantiation-checker/complex_1.move:13:5
+   │
+13 │ ╭     fun c<T1, T2>() {
+14 │ │         c<u64, T1>();
+15 │ │         d<T2>();
+16 │ │         e<T2>()
+17 │ │     }
+   │ ╰─────^
+   │
+   = c<T1, T2> calls d<#1> at tests/recursive-instantiation-checker/complex_1.move:13
```

Comment:
This chain of instantiations is clearer than what I see from V1, but it also doesn't quite make sense.  Can we express it succinctly as "A call to `c<T1, T2>` leads to a call to `c<M::S<T2>, bool>`" which leads to increasing type sizes?

But as stated, I seems like the next call in that sequence is `c<M::S<bool>, bool>` which seems to not be expanding.

Reply:
The expansion is correct, even though some types are not displayed correctly. The trace is important for debug. I added the v1 notes on " recursive calls causes the instantiation to recurse infinitely". But I think the best we do is to document the term "cyclic type instantiation" in the move book, and reserve an error code for this.